src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.exceptions.CallbackException;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackHandler;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackParams;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.GenerateOrderCallbackHandler;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.MoreTimeRequestedCallbackHandler;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.GENERATE_ORDER;
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.MORE_TIME_REQUESTED_PAPER;
+
+@Component
+public class CallbackHandlerFactory {
+
+    private Map<String, CallbackHandler> handlers;
```

Comment:
I opted for having a `Map<String, CallbackHandler>` rather then the proper enum there because this way I can just handle the case of a wrong/unsupported enum in the dispatch method, from a dev point of view nothing really changes